libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -3710,6 +3711,20 @@ MAV_RESULT GCS_MAVLINK::handle_command_do_gripper(const mavlink_command_long_t &
     return result;
 }
 
+#if HAL_SPRAYER_ENABLED
+MAV_RESULT GCS_MAVLINK::handle_command_do_sprayer(const mavlink_command_long_t &packet)
+{
+    AC_Sprayer *sprayer = AP::sprayer();
+    if (sprayer == nullptr) {
+        return MAV_RESULT_FAILED;
+    }
+
+    sprayer->run(uint8_t(packet.param1));
```

Comment:
run takes a boolean.. I'm sure this will work but I guess I would make it "is_positive(packet.param1)"

Reply:
I might just cast this to a boolean instead.  I'd like to keep non-boolean values Reserved For Future Use, if that's OK.

Reply:
I've adjusted this and the mission equivalent to strictly take 0 or 1 to allow for future modifications to the use of this value.